frontend/libs/studio-components/src/components/StudioPageHeader/StudioPageHeaderProfileMenu/types/StudioProfileMenuItem.ts

```diff
@@ -0,0 +1,16 @@
+type StudioProfileMenuItemButton = {
+  type: 'button';
+  onClick: () => void;
+};
+
+type StudioProfileMenuItemLink = {
+  type: 'link';
+  href: string;
+  openInNewTab?: boolean;
+};
+
+export type StudioProfileMenuItem = {
+  action: StudioProfileMenuItemButton | StudioProfileMenuItemLink;
+  itemName: string;
+  isActive?: boolean;
+};
```

Comment:
Dette er en fin mulighet til å utnytte [generiske typer](https://www.typescriptlang.org/docs/handbook/2/generics.html):
```suggestion
export type StudioProfileMenuItem<T extends StudioProfileMenuItemType = StudioProfileMenuItemType> =
  {
    action: StudioProfileMenuItemAction<T>;
    itemName: string;
    isActive?: boolean;
  };

type StudioProfileMenuItemType = 'button' | 'link';

type StudioProfileMenuItemAction<T extends StudioProfileMenuItemType> = {
  button: StudioProfileMenuItemButtonAction;
  link: StudioProfileMenuItemLinkAction;
}[T];

type StudioProfileMenuItemButtonAction = {
  type: 'button';
  onClick: () => void;
};

type StudioProfileMenuItemLinkAction = {
  type: 'link';
  href: string;
  openInNewTab?: boolean;
};
```
Nå kan vi henvise til de to forskjellige typene ved å bruke `StudioProfileMenuItem<'button'>` og `StudioProfileMenuItem<'link'>`. Dette kommer til nytte i de to forskjellige komponentene.